core/src/test/scala/unit/kafka/log/remote/RemoteIndexCacheTest.scala

```diff
@@ -1086,4 +1087,10 @@ class RemoteIndexCacheTest {
       case e @ (_ : NoSuchFileException | _ : UncheckedIOException) => Optional.empty()
     }
   }
+
+  private def numThreadsRunning(threadNamePrefix: String, isDaemon: Boolean): mutable.Set[Thread] = {
```

Comment:
We can fix the build error by rewriting this function via java lambda. For example:
```scala
  private def numThreadsRunning(threadNamePrefix: String, isDaemon: Boolean): java.util.Set[Thread] = {
    Thread.getAllStackTraces.keySet.stream().filter { t =>
      isDaemon && t.isAlive && t.getName.startsWith(threadNamePrefix)
    }.collect(Collectors.toSet)
  }
```

Reply:
I change the function name, and modify using java set
```java  
private def getRunningCleanerThread: java.util.Set[Thread] = {
    Thread.getAllStackTraces.keySet()
      .stream()
      .filter(t => t.isAlive && t.getName.startsWith(REMOTE_LOG_INDEX_CACHE_CLEANER_THREAD))
      .collect(Collectors.toSet())
  }
  ```